### PR TITLE
Warning external task updated to append warnings

### PR DESCRIPTION
**Pull request by indvamshi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-604

### Change description ###

- Warning external task updated to append warnings to matching process instance ids

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/handler/WarningTaskWorkerHandler.java [hunk 1]
```diff
@@ -3,29 +3,35 @@
...
+        externalTaskService.complete(externalTask, Map.of(
+            "hasWarnings",
+            true,
+            "warningList",
+            mapWarningValues(variables)
+        ));
     }
+
+    private String mapWarningValues(Map<?, ?> variables) {
+        final String warningStr = (String) variables.get("warningList");
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/handler/WarningTaskWorkerHandler.java:26]: could it happen that warningList is not present and then warningStr to be null?

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/handler/WarningTaskWorkerHandler.java [hunk 2]
```diff
@@ -3,29 +3,35 @@
...
+        WarningValues warningValues = new WarningValues(warningStr);
+
+        Warning warning = new Warning((String) variables.get("warningCode"),
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/handler/WarningTaskWorkerHandler.java:29]: same here, we are not checking warning could be empty or null, is this correct?

**Review by iguannago — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/model/WarningValues.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        return values;
+    }
+
+    public String getValuesAsJson() {
+        try {
+            return new ObjectMapper().writeValueAsString(values);
+        } catch (JsonProcessingException jsonProcessingException) {
+            log.error("Could not deserialize Warning values");
+        }
+        return StringUtils.EMPTY;
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/model/WarningValues.java:51]: add unit test for getValueAsJson returning **StringUtils.EMPTY**?

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/handler/WarningTaskWorkerHandler.java [hunk 4]
```diff
@@ -1,31 +1,59 @@
...
+        String caseId = (String) variables.get("caseId");
+        log.info("caseId {} and its warning values : {}", caseId, warningValues.getValuesAsJson());
+        return warningValues.getValuesAsJson();
+    }
+
+    private Warning mapWarningAttributes(Map<?, ?> variables) {
+        final String warningCode = (String) variables.get("warningCode");
+        final String warningText = (String) variables.get("warningText");
+
+        if (warningCode != null && warningText != null) {
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/handler/WarningTaskWorkerHandler.java:52]: add unit test to cover when this return null?

Comment (indvamshi) [reply src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/handler/WarningTaskWorkerHandler.java:63]: I have checked, it does cover the null.

**Review by BenNzewi — APPROVED** [review]
